src/Functions/jsonMergePatch.cpp

```diff
@@ -102,8 +102,11 @@ namespace
                     throw Exception(ErrorCodes::BAD_ARGUMENTS, "Wrong JSON string to merge. Expected JSON object");
             };
 
-            const auto * first_string = typeid_cast<const ColumnString *>(arguments[0].column.get());
-            if (!first_string)
+            const auto * first_column_arg = arguments[0].column.get();
+            const auto * first_column_arg_string = typeid_cast<const ColumnString *>(first_column_arg);
+            const auto * first_column_arg_const = typeid_cast<const ColumnConst *>(first_column_arg);
```

Comment:
Should we also check that nested column is String?

Reply:
I would do smth like this:
```
bool is_first_const = isColumnConst(*arguments[0].column);
const auto * first_column_arg_string = is_first_const
            ? checkAndGetColumnConstData<ColumnString>(arguments[0].column.get())
            : checkAndGetColumn<ColumnString>(arguments[0].column.get());
```
And later pass always 0 index to `parse_json_document` if `is_first_const=true`

Reply:
Well, now we pass not `ColumnString` to the lambda, but `IColumn`, so we call virtual method `getDataAt`, and devirtualization on each row can be expensive

Reply:
I may be wrong, if you think current way is ok, then no problem